### ISSUE TITLE
Need API to verify if ARCore is installed or not

### ISSUE THREAD

**Issue by jayshildave** [google-ar/arcore-android-sdk#640]
Need API to verify just the install status not go ahead and request install.
Is there any API available which can help in understand whether it's currently installed or not.

**Comment by jayshildave** [comment]
Should have read the documentation :( 

public ArCoreApk.InstallStatus requestInstall (Activity applicationActivity, boolean **userRequestedInstall**)
When your activity resumes, you should call this method again, this time with userRequestedInstall = false. This will either return INSTALLED or throw an exception indicating the reason that installation could not be completed. I am assuming this will not request install just provide status

**Comment by jayshildave** [comment]
It only works the second time doesn't work first time. If we need to have some information to be provided to the user with a choice whether they wish to play the AR Content or not. Thats not possible. Would need to have a way of verifying if the ARCore is installed or not to show the user some context before the installation process begins.

**Comment by jayshildave** [comment]
Not really ideal but as a work around for now. 
`packageManager.getPackageInfo("com.google.ar.core", 0);`

Risk if the package name changes it goes for the toss.

**Comment by inio** [comment]
Sorry for the slow response on this.

`ArCoreApk.checkAvailability()` can be used to determine the current support for ARCore.  Note that if ARCore is not currently installed, it will return `UNKNOWN_CHECKING` for some time before having a final code.  On devices without network access or that are not logged in to the Play Store, it may return `UNKNOWN_CHECKING` indefinitely.

**Comment by jayshildave** [comment]
@inio thats correct we use that "Availability" to identify whether or not AR is supported on a particular device.
 
The question i need to know is if the AR Core Package is installed or not. Which can only be understood when we do requestInstall which will return back INSTALLED if its present or will return back INSTALLATION_REQUESTED. 

We want to provide some context to user when they would see the AR Core Installation message. We do that by checking if AR Core Package was install where we inform the user that they would be seeing a AR Core Installation dialog and they should install it. If they wish not to install it we have something else to provide them. 

So having a API which will just check for the INSTALLATION status and not begin requesting installation the user would be helpful. As that would help us provide some more context to the user what they are going to view ahead (AR Core Installation Dialog and so on and so forth.)

**Comment by inio** [comment]
`checkAvailability()` will return `SUPPORTED_INSTALLED` if ARCore is ready and `requestInstall()` will be a noop.  If it returns `SUPPORTED_NOT_INSTALLED` or `SUPPORTED_APK_TOO_OLD` then you'll get an installation prompt when you call `requestInstall()`
